Adding placement and span options

christianfares opened the pull request:
- Added option and examples for a span layout which spans the annotation across the region rather than put a marker in the center of the region
- Added a placement option to allow annotations to be placed on either side of the ideogram, and to allow annotations to stack on top of each other.

eweitz commented:
Thanks for this, @christianfares and @chaisarina!  Below are two requests to ease review and maintenance, and help others understand the impact of your work.  Please edit your original PR description above to add:

* Screenshot images and/or [screen recording](https://www.google.com/search?q=how+to+record+your+screen) videos show how this looks
* Explanation of biological use case

eweitz reviewed (commented):
Thanks for this, @christianfares and @chaisarina!  Below are two requests to ease review and maintenance, and help others understand the impact of your work.  Please edit your original PR description above to add:

* Screenshot images or [screen recording](https://www.google.com/search?q=how+to+record+your+screen) videos show how this looks
* Explanation of biological use case

christianfares commented:
Hi @eweitz, sorry for the blated response, I was on leave for 2 weeks.

We currently use this package in a in-house built genomic visualization portal where our researchers have access to different sources of data, such copy number variation, RNA, single nucleotide and short indel variations etc. We wanted to to show all this data on a single plot. So we have added the 'span' to show copy number + LOH on a region on the chromosome, and the histo option to show RNA expression levels. This would allow researchers to see correlation between the data more easily. We also show Pathogenic SNVs using the bre-built triangle shape. Here is an example (using some fake data):
<img width="1010" alt="image" src="https://github.com/user-attachments/assets/4722e7e2-688f-464e-a17b-07d88bab8c53" />

Here the green shows low copy number (< 2), red is high copy number (> 2), grey is neutral (~ 2). The purple indicates LOH, and the histogram is the RNA expression.

This can be achieved using the following track configuration:
```
    const annotationTracksH = [
      {
        id: 'cnTrack', displayName: 'Copy Number', shape: 'span', placement: 0,
      },
      {
        id: 'lohTrack', displayName: 'LOH', shape: 'span', placement: 1,
      },
      {
        id: 'reportableSnvs', displayName: 'Reportable SNVs', shape: 'triangle', placement: 2,
      },
      {
        id: 'rnaexpressionLevelTrack',
        displayName: 'RNA-Seq Expression level',
        shape: 'histo',
        placement: -2,
      }, // WITH OUTLIERS
    ];

    const heatmaps = [
      {
        key: 'cn',
        thresholds: [
          ['-1', 'rgba(0,0,0,0)'],
          ['-0.1', '#0EC971'],
          ['1', '#5DFCA8'],
          ['1.7', '#C9FFE2'],
          ['2.5', '#ECF0F3'],
          ['4', '#FEE0E9'],
          ['6', '#FF85A9'],
          ['+', '#FF2969'],
        ],
      },
      {
        key: 'loh',
        thresholds: [
          ['0', 'rgba(0,0,0,0)'],
          ['1', '#d19cff'],
        ],
      },
      {
        key: 'pathclass',
        thresholds: [
          ['-1', 'rgba(0,0,0,0)'],
          ['0', '#FF2969'],
          ['1', '#FF85A9'],
          ['2', '#5DFCA8'],
          ['3', '#5DFCA8'],
          ['4', '#D0D9E2'],
          ['5', '#D0D9E2'],
          ['6', '#D0D9E2'],
          ['7', '#D0D9E2'],
        ],
      },
      {
        key: 'rna',
        thresholds: [
          ['0', 'purple'],
          ['+', 'purple'],
        ],
      },
    ];

    const histogram = [
      {
        key: 'rna',
        thresholds: [
          ['0', '0'],
          ['1', '4'],
          ['2', '8'],
          ['3', '12'],
          ['4', '16'],
          ['5', '20'],
        ],
      },
    ];

    const makeChromosome = (ref: HTMLDivElement): Ideogram => new Ideogram({
      organism: 'human',
      assembly: 'GRCh38',
      orientation: 'horizontal',
      chromosome: chr,
      showChromosomeLabels: false,
      chrHeight: ref.clientWidth * 0.9,
      chrWidth: width,
      chrMargin: 0,
      container: `#${ref.id}`,
      showBandLabels: true,
      barWidth: 4,
      heatmaps,
      histogram,
      annotationHeight: view === 'modal' ? 6 : 10,
      annotationTracks: annotationTracksH,

      annotations: {
        keys: ['name', 'start', 'length', 'cn', 'loh', 'pathclass', 'rna'], // add rnaexp
        annots: [{ chr, annots: annotations || [] }],
      },
    });
```

We have added an example also in `examples/vanilla/annotations-span-heatmap-histogram.html`

eweitz commented:
This looks great, @christianfares and @chaisarina!  I just merged it.  

Two questions:

1. Which organization are you using these enhancements with?  Knowing who's using Ideogram helps maintenance.
2. What name and email address would you like to be added to the [AUTHORS](https://github.com/eweitz/ideogram/blob/master/AUTHORS) file?  By default I'll use information from your GitHub profile and email address from Git logs.

christianfares commented:
Thanks @eweitz!

We work for the Children's Cancer Institute (https://www.ccia.org.au) and we use the package in an internal platform that our researchers use when reviewing patient data.

For our names in the AUTHORS file, could you please list the our name with the institute name like this:
Name (Children's Cancer Institute) \<email\>.
For me it would be: Christian Fares (Children's Cancer Institute) \<cfares@ccia.org.au\>.
